nucleus/glassfish-jul-extension/src/main/java/org/glassfish/main/jul/handler/LogRecordBuffer.java

```diff
@@ -43,10 +45,12 @@
  */
 class LogRecordBuffer {
 
-    private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock(true);
-    private int capacity;
-    private int maxWait;
-    private ArrayBlockingQueue<GlassFishLogRecord> pendingRecords;
+    private final BlockingQueue<GlassFishLogRecord> pendingRecords = new LinkedBlockingQueue<>();
```

Comment:
Doesn't have linked blocking queue significantly lower performance than ArrayBlockingQueue? I have to try ...

Reply:
Seems it is the opposite, hmmm!

> It has a higher throughput than array-based queues.